src/main/java/com/online/store/dao/jdbc/impl/ProductDaoJdbcImpl.java

```diff
@@ -0,0 +1,116 @@
+package com.online.store.dao.jdbc.impl;
+
+import com.online.store.dao.ProductDao;
+import com.online.store.exceptions.DataProcessingException;
+import com.online.store.lib.Dao;
+import com.online.store.model.Product;
+import com.online.store.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ProductDaoJdbcImpl implements ProductDao {
+    @Override
+    public Product create(Product item) {
+        String query = "INSERT INTO products (name, price) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            Product product = new Product(item.getName(), item.getPrice());
+            PreparedStatement statement = connection
+                    .prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            statement.setString(1, item.getName());
+            statement.setDouble(2, item.getPrice());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                product.setId(resultSet.getLong(1));
+            }
+            return product;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create product", e);
+        }
+    }
+
+    @Override
+    public Product update(Product item) {
+        String query = "UPDATE products SET name = ?, price = ? WHERE product_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setString(1, item.getName());
+            statement.setDouble(2, item.getPrice());
+            statement.setLong(3, item.getId());
+            int numberOfUpdatedProducts = statement.executeUpdate();
+            if (numberOfUpdatedProducts != 0) {
+                return item;
+            }
+            throw new SQLException();
```

Comment:
it's  better to throw our custom exception and add some message

Reply:
done